services/QuillLMS/client/app/bundles/Teacher/components/classrooms/import_provider_classroom_students_modal.tsx

```diff
@@ -0,0 +1,107 @@
+import * as React from 'react';
+
+import { requestPut } from '../../../../modules/request/index';
+import pusherInitializer from '../../../../modules/pusherInitializer'
+import ButtonLoadingIndicator from '../shared/button_loading_indicator';
+import { providerLookup } from './providerHelpers'
+
+const smallWhiteCheckSrc = `${process.env.CDN_URL}/images/shared/check-small-white.svg`
+
+interface ImportProviderClassroomStudentsModalProps {
+  classroom: any;
+  close: () => void;
+  onSuccess: (snackbarCopy: string) => void;
+  provider: string
+  user: any
+}
+
+const ImportProviderClassroomStudentsModal = ({ close, classroom, onSuccess, provider, user }: ImportProviderClassroomStudentsModalProps) => {
+  const providerTitle = providerLookup[provider].title
+  const providerClassName = providerLookup[provider].className
+  const importStudentsPath = `/${providerClassName}_integration/teachers/import_students`
+  const importStudentsEventName = `${providerClassName}-classroom-students-imported`
+
+  const [termsAccepted, setTermsAccepted] = React.useState(false);
+  const [waiting, setWaiting] = React.useState(false);
+
+  const handleImportStudents = () => {
+    setWaiting(true)
+
+    pusherInitializer(user.id, importStudentsEventName, () => onSuccess('Class re-synced'))
+    requestPut(importStudentsPath, { classroom_id: classroom.id })
```

Comment:
I think this handles the race condition we discovered yesterday, right? Thanks for doing that.